truetype/truetype.go

```diff
@@ -240,6 +260,48 @@ func (f *Font) parseCmap() error {
 			offset += 2
 		}
 		f.cmapIndexes = f.cmap[offset:]
+
+		// If unable to parse cmap correctly, use unitype cmap parse implementation.
+		if len(f.cmapIndexes) < 1 {
```

Comment:
so we have two implementations for this? is that necessary?

Reply:
Yes, that cmapIndexes is being used at
`func (f *Font) Index(x rune) Index `
```
for i, j := 0, len(f.cm); i < j; {
		h := i + (j-i)/2
		cm := &f.cm[h]
		if c < cm.start {
			j = h
		} else if cm.end < c {
			i = h + 1
		} else if cm.offset == 0 {
			return Index(c + cm.delta)
		} else {
			offset := int(cm.offset) + 2*(h-len(f.cm)+int(c-cm.start))
			return Index(u16(f.cmapIndexes, offset))
		}
	}
```
and that part is used for format12 and if the format4 cmap can be parsed correctly without the unitype implementation.